operations.go

```diff
@@ -0,0 +1,75 @@
+package webrtc
+
+import (
+	"sync"
+)
+
+// Operation is a function
+type operation func()
+
+// Operations is a task executor.
+type operations struct {
+	ops     []operation
+	mu      sync.Mutex
+	startMu sync.Mutex
+	// closed channel is alway closed and is returned when there are no queued
+	// operations.
+	closed chan struct{}
+	done   chan struct{}
+}
+
+func newOperations() *operations {
+	closed := make(chan struct{})
+	close(closed)
+	return &operations{
+		closed: closed,
```

Comment:
Sorry I mislead you in my last comment, what about `&operations{ done : closed}`? Then you can drop the nil check!

Reply:
Haha that makes perfect sense - done! Only one `if` clause in the whole implementation now 😎